viewer/cache.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef HAWKTRACER_VIEWER_CACHE_HPP
+#define HAWKTRACER_VIEWER_CACHE_HPP
+
+#include <hawktracer/base_types.h>
+#include <hawktracer/parser/event.hpp>
+
+#include <unordered_map>
+#include <vector>
+
+namespace HawkTracer
+{
+namespace viewer
+{
+
+using EventRef = std::reference_wrapper<const parser::Event>;
+
+class Cache
```

Comment:
Could you please add unittests for this class and for `EventDB` class so we can make sure that cache works as expected?